### PR TITLE
docs: add broadcast cheatcodes

### PR DESCRIPTION
**Pull request by shawnharmsen**
closes: https://github.com/foundry-rs/book/issues/371

### CODE REVIEW

File: src/cheatcodes/broadcast.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```solidity
+function broadcast(address who) external;
+```
+
+### Description
+
+Using the address that calls the test contract or the address provided
+as the sender, has the next call (at this call depth only) create a
+transaction that can later be signed and sent onchain.
```
Comment (onbjerg) [inline src/cheatcodes/broadcast.md:17]:
```suggestion
Using the address that calls the test contract or the address provided
as the sender, have the next call (at this call depth only) create a
transaction that can later be signed and sent onchain.
```

File: src/cheatcodes/broadcast.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Examples
+
+```solidity
+function deploy() public {
+    cheats.broadcast(ACCOUNT_A);
+    Test test = new Test();
+
+    // this wont generate tx to sign
```
Comment (onbjerg) [inline src/cheatcodes/broadcast.md:26]:
```suggestion
    // this won't generate tx to sign
```

File: src/cheatcodes/start-broadcast.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+```solidity
+function startBroadcast(address who) external;
+```
+
+### Description
+
+Using the address that calls the test contract or the address provided
+as the sender, has all subsequent calls (at this call depth only) create
+transactions that can later be signed and sent onchain.
```
Comment (onbjerg) [inline src/cheatcodes/start-broadcast.md:17]:
```suggestion
Using the address that calls the test contract or the address provided
as the sender, have all subsequent calls (at this call depth only) create
transactions that can later be signed and sent onchain.
```

File: src/cheatcodes/start-broadcast.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    function deployMany() public {
+        cheats.startBroadcast();
+
+        for(uint i; i< 100; i++) {
+            NoLink test9 = new NoLink();
+        }
+
+        cheats.stopBroadcast();
+    }
+}
```
Comment (onbjerg) [inline src/cheatcodes/start-broadcast.md:46]: I think this example is very hard to parse, we should find something simpler - since we're just demoing deployment, we probably should just have a very small contract with clear functions, and without reference to linking at all. This example also uses DSTest, not Forge Std

File: src/cheatcodes/stop-broadcast.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+## `stopBroadcast`
+
+### Signature
+```solidity
+function stopBroadcast() external;
+```
+
+### Description
+
+Stops collecting onchain transactions.
```
Comment (onbjerg) [inline src/cheatcodes/stop-broadcast.md:10]:
```suggestion
Stops collecting transactions for later on-chain broadcasting.
```

File: src/cheatcodes/stop-broadcast.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+### Examples
+
+```solidity
+function deployNoArgs() public {
+    cheats.broadcast();
+    Test test1 = new Test();
+
+    cheats.startBroadcast();
+    Test test2 = new Test();
+    cheats.stopBroadcast();
```
Comment (onbjerg) [inline src/cheatcodes/stop-broadcast.md:21]:
```suggestion
    // broadcast the next call
    cheats.broadcast();
    Test test1 = new Test();

    // broadcast all calls between this line and `stopBroadcast`
    cheats.startBroadcast();
    Test test2 = new Test();
    cheats.stopBroadcast();
```

**Review by onbjerg — CHANGES_REQUESTED** [review]
Some minor things - wording and examples

**Comment by shawnharmsen** [comment]
Let me know if the changed example for startBroadcast is better

was getting everything from here: https://github.com/foundry-rs/foundry/blob/master/testdata/cheats/Broadcast.t.sol

```solidity
function t(uint256 a) public returns (uint256) {
    uint256 b = 0;
    emit log_string("here");
    return b;
}
function deployOther() public {
    cheats.startBroadcast(ACCOUNT_A);
    Test test = new Test();
    
    // will trigger a transaction
    test.t(1);
    
    cheats.stopBroadcast();
}

**Review by onbjerg — APPROVED** [review]
I think this is good, we can always iterate if needed :) Thanks!
